.github/workflows/ci-backend-cql.yml

```
@@ -0,0 +1,91 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: CI Backend CQL
+
+on:
+  pull_request:
+    paths-ignore:
+    - 'docs/**'
+    - '.github/workflows/ci-docs.yml'
+    - 'requirements.txt'
+    - 'docs.Dockerfile'
+    - '*.md'
+  push:
+    paths-ignore:
+    - 'docs/**'
+    - '.github/workflows/ci-docs.yml'
+    - 'requirements.txt'
+    - 'docs.Dockerfile'
+    - '*.md'
+
+env:
+  BUILD_MAVEN_OPTS: "-DskipTests=true --batch-mode --also-make"
+  VERIFY_MAVEN_OPTS: "-Pcoverage"
+
+jobs:
+  build-all:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - run: mvn clean install --projects janusgraph-all -Pjanusgraph-cache -Dmaven.javadoc.skip=true ${{ env.BUILD_MAVEN_OPTS }}
+      - run: mvn verify --projects janusgraph-all -Pjanusgraph-cache ${{ env.VERIFY_MAVEN_OPTS }}
+
+  tests:
+    runs-on: ubuntu-20.04
+    needs: build-all
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - module: cql
+            args: "-Pcassandra3-byteordered -Dtest=\"**/diskstorage/cql/*\""
+          - module: cql
+            args: "-Pcassandra3-murmur -Dtest=\"**/diskstorage/cql/*\""
+          - module: cql
+            args: "-Pcassandra3-byteordered -Dtest=\"**/graphdb/cql/*\""
+          - module: cql
+            args: "-Pcassandra3-murmur -Dtest=\"**/graphdb/cql/*\""
+          - module: cql
+            args: "-Pcassandra3-murmur -Dtest=\"**/hadoop/*\""
+          - module: cql
+            args: "-Pcassandra3-murmur-ssl -Dtest=\"**/diskstorage/cql/CQLStoreTest.java\""
+          - module: cql
+            args: "-Pcassandra3-murmur-client-auth -Dtest=\"**/diskstorage/cql/CQLStoreTest.java\""
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - run: mvn clean install --projects janusgraph-${{ matrix.module }} ${{ env.BUILD_MAVEN_OPTS }} ${{ matrix.install-args }}
+      - run: mvn verify --projects janusgraph-${{ matrix.module }} ${{ env.VERIFY_MAVEN_OPTS }} ${{ matrix.args }}
+      - name: cql 3.0.18
+        if: "contains(github.event.head_commit.message, '[cql-tests]')"
```

What about following the same rule as [tp-tests]? 
```suggestion
        if: "github.event_name == 'push' && contains(github.event.head_commit.message, '[cql-tests]') || github.event_name == 'pull_request' && contains(github.event.pull_request.title, '[cql-tests]')"

```
and in RELEASING.md, shall we mention [cql-tests] in the _Create a release commit_ section as well?

---

You may even make [full build] = [cql-tests] + [tp-tests] + (just in case more compatibility test suites added in the future)

---

I would move the full build stuff to a next PR. 
I will update check for CQL. 
I will try add some sentence to the releasing section.

---

https://github.com/JanusGraph/janusgraph/issues/2302